modules/postgresql/variables.tf

```
@@ -34,6 +34,11 @@ variable "random_instance_name" {
 variable "database_version" {
   description = "The database version to use"
   type        = string
+
+  validation {
+    condition     = (length(var.database_version) >= 9 && ((substr(var.database_version, 0, 9) == "POSTGRES_" || substr(var.database_version, 0, 9) == "postgres_") && can(regex("^\\d+(?:_?\\d)*$", substr(var.database_version, 9, -1))))) || can(regex("^\\d+(?:_?\\d)*$", var.database_version))
```

nit: I think we can just lower case to keep this shorter
```suggestion
    condition     = (length(var.database_version) >= 9 && (lower(substr(var.database_version, 0, 9) && can(regex("^\\d+(?:_?\\d)*$", substr(var.database_version, 9, -1))))) || can(regex("^\\d+(?:_?\\d)*$", var.database_version))
```

---

I have made the changes based on your suggestion.

I have also added "replace(var.database_version, substr(var.database_version, 0,8), "POSTGRES")" instead of just "var.database_version", to avoid terraform drift. (This drift was because "postgres_14" was being converted to "POSTGRES_14" by the API, hence there was a difference in state file and config file, hence the fix.)